sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -177,28 +188,41 @@ impl CosmosClient {
 
         let auth = {
             let resource_link = generate_resource_link(&uri_path);
+            trace!(
+                "resource_link generated by prepare_request == {}",
+                resource_link
+            );
             generate_authorization(
                 &self.auth_token,
                 &http_method,
-                resource_type,
+                &resource_type,
                 resource_link,
                 &time,
             )
         };
         self.prepare_request_with_signature(uri_path, http_method, &time, &auth)
     }
 
-    /// Prepares' an `azure_core::Request`.
+    /// Prepares' an `azure_core::Request`. This function will
```

Comment:
nit: not sure if we need to make the comment lines so short. Means we quickly have lots of lines of comments.

Reply:
I have a vim plugin that inserts a cr to keep comments < 80 chars wide (yes, I'm old 🤷‍♂️). I agree it doesn't really matter since Rust's lines can easily go over that limit. What should we use as a limit?